File: src/menu/components/TeamListPanel.java
package menu.components;

import java.util.LinkedList;

import menu.Menu;
import menu.ComponentContainer;
import menu.NetworkingMenu;
import menu.MenuComponent;
import menu.components.Panel;
import menu.components.TeamPanel;
import menu.menues.LobbyMenu;
import misc.game.Team;
import misc.math.Rect;
import network.lobby.LobbyPlayer;

public class TeamListPanel extends Panel
{
	public TeamListPanel(ComponentContainer parent, Rect rect)
	{
		super(parent, rect);

		getComponents().add(new TeamPanel(this, new Rect(20, 20, 140, 60), Team.TEAM0));
		getComponents().add(new TeamPanel(this, new Rect(20, 120, 140, 60), Team.TEAM1));
		getComponents().add(new TeamPanel(this, new Rect(20, 220, 140, 60), Team.TEAM2));
		getComponents().add(new TeamPanel(this, new Rect(20, 320, 140, 60), Team.TEAM3));
		getComponents().add(new TeamPanel(this, new Rect(20, 420, 140, 60), Team.TEAM4));
		update(); // erstellt die nötigen Player icons, etc...
	}

	// aktualisiert die PlayerIcons; Wird von den LobbyMenüs aufgerufen
	public void update()
	{
		// Formatieren der TeamPanels (nach unten rutschen, falls das obere Panel zu weit nach unten reicht)
		for (int i = 0; i < getComponents().size(); i++) // für alle TeamPanels updaten + formatieren
		{
			((TeamPanel) getComponents().get(i)).update(getPlayers()); // updaten
			((Panel) getComponents().get(i)).calcSize();
			if (i != 0) // Das oberste Panel, wird nie nach unten gerückt, weil das ja logisch ist
			{
				getComponents().get(i).setPosition(20, getComponents().get(i-1).getBot() + 20);
			}
		}
	}

	// Setter
	public void disableTeamButtons()
	{
		for (int i = 0; i < getComponents().size(); i++)
		{
			((TeamPanel) getComponents().get(i)).disableTeamButton();
		}
	}

	// Getter
	private LinkedList<LobbyPlayer> getPlayers()
	{
		return ((LobbyMenu) getParent()).getPlayers();
	}
}

File: cpp/src/menu/components/TeamListPanel.cpp
#include "TeamListPanel.hpp"

#include <misc/Global.hpp>
#include "TeamPanel.hpp"
#include <team/Team.hpp>
#include <menu/menues/LobbyMenu.hpp>

TeamListPanel::TeamListPanel(LobbyMenu* c, const PixelRect& r) : Panel(c, r), lobby(c)
{
	update();
}

void TeamListPanel::update()
{
	// Formatieren der TeamPanels (nach unten rutschen, falls das obere Panel zu weit nach unten reicht)
	clearComponents();
	for (unsigned int i = 0; i < Team::getAmount(); i++) // für alle TeamPanels updaten + formatieren
	{
		if (i == 0)
		{
			TeamPanel* panel = new TeamPanel(getLobbyMenu(), this, PixelRect(20, 20+100*i, 140, 60), Team::get(i));
			addComponent(panel);
			teamPanels.push_back(panel);
		}
		else
		{
			TeamPanel* panel = new TeamPanel(getLobbyMenu(), this, PixelRect(20, teamPanels[i-1]->getRelativeRect().getBot() + 20, 140, 60), Team::get(i));
			addComponent(panel);
			teamPanels.push_back(panel);
		}
		teamPanels[i]->update(getLobbyMenu()->getPlayers()); // updaten
		teamPanels[i]->calcSize();
	}
}

LobbyMenu* TeamListPanel::getLobbyMenu() const
{
	return lobby;
}

void TeamListPanel::disableTeamButtons()
{
	for (unsigned int i = 0; i < teamPanels.size(); i++)
	{
		teamPanels[i]->disableTeamButton();
	}
}
